src/audio/kpb.c

```diff
@@ -91,6 +91,10 @@ struct comp_data {
 #ifdef CONFIG_IPC_MAJOR_4
 	struct ipc4_kpb_module_cfg ipc4_cfg;
 #endif /* CONFIG_IPC_MAJOR_4 */
+	uint32_t mic_sel_cnt_;
+	size_t num_of_in_channels_;
+	uint16_t offsets_[4];
```

Comment:
this array is assigned but I don't see it being used in this commit

Reply:
Also, I tried to find logic in type selection but I couldn't. Why is number of microphones `uint32_t` and number of channels `size_t`? Why are offsets 16-bit? Do we expect more than 255 microphones?

Reply:
,,offsets" is used in kpb_set_micselect function

Reply:
Hmm @lyakh array is used in set_large_config ?

I wonder about the "[4]". Maybe this should be a define and there probably should be some comment why it is 4.

Reply:
Done

Reply:
> Hmm @lyakh array is used in set_large_config ?
> 
> I wonder about the "[4]". Maybe this should be a define and there probably should be some comment why it is 4.

@kv2019i I said "in this commit" - yes, it's used in the second commit. It's ok but just a bit confusing. If possible it's usually good to add data when it's needed, but it isn't critical of course. Within one PR it's still ok.

Reply:
> ,,offsets" is used in kpb_set_micselect function. What size would be appropriate?

I think you could use one of two approached in this and similar cases: (1) tight optimisation for space, you really use types that are enough for values they might need to contain. Then you could use `unsigned char` for offsets, etc. Or (2) you use "natural" types that should be the easiest for the DSP to handle, that are best suitable for integer operations - then I'd just use `int` or `unsigned int` or `size_t` or `uintptr_t` as appropriate. Using `uint16_t` immediately creates an impression, that it's important for this field to have that type - either for compatibility with other types, e.g. if a pointer to it is used as a function argument, or because 8 bits aren't enough and 16 bits are guaranteed to be enough and we are tight on space, none of which seem to hold.

Reply:
Done